Supply missing references to USE_VG_DRD macro

GBuella opened the pull request:
Not checking for the USE_VG_DRD macro in src/common/util.c can cause a
naive developers build using -DUSE_VG_DRD to fail.
Since src/common/out.c refers to the variable _On_valgrind when
USE_VG_DRD is defined, the said variable shall be defined in util.c

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/758)
<!-- Reviewable:end -->

krzycz commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/pmem/nvml/758#-KFDUJOjvGtF7L6d_RlZ:-KFDUJOkXVqN-KmT1fll:-2088287902) (no related file):*
Good catch! But you also need to update some *.match files. Otherwise, unit tests would fail when run under Valgrind.

BTW, perhaps we could modify the implementation to have just one message in the log, like:
"compiled with support for Valgrind: memcheck, drd, helgrind"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/758)*
<!-- Sent from Reviewable.io -->

GBuella commented:
I updated the related .match files.
While at it, I added a suppression file, to avoid false positives from helgrind ( the same ones about fputs as one gets from drd ). With this, I finally managed to run make check successfully. I have no idea why this didn't come up with the travis builds, it might just be luck.

GBuella commented:
So the travis build fails again. I guess traces_pmem/trace_pmem0.log.match didn't expect this many log lines.
I can't run it with -DUSE_VALGRIND , since I don't have pmemcheck. Maybe I should get it..

BTW there is one more minor issue with the valgrind annotation: out_print_func in src/common/out.c uses the _On_valgrind variable ( via the annotation macros ), which is initalized in util.c . But out_print_func is also called before util.c's initialization function is called, which means it can falsely assume that it is not running under valgrind. I confirmed it by simply printing the _On_valgrind variable in out_print_func. Under valgrind, it is seen as zero in the first few calls.

marcinslusarz commented:
Can you put this patch on hold for a few days? @ryspol has a big patch that reworks how Valgrind is enabled, simplifies success checks and as a side effect is going to fix some of the issues you pointed out.

marcinslusarz commented:
All pending patches related to Valgrind were merged to master. Can you update this pull request?

Please follow commit message conventions described here: http://pmem.io/2014/09/09/git-workflow.html (in "Developing and merging locally" section).

GBuella commented:
Looking at the valgrind related I changes, I rather just close this request.
